lib/execution/runner.js

```diff
@@ -179,13 +182,13 @@ class Runner {
 
         let cancelQuery;
         if (obj.cancelOnTimeout) {
-          cancelQuery = this.client.cancelQuery(this.connection);
+          cancelQuery = this.client.cancelQuery(connection);
         } else {
           // If we don't cancel the query, we need to mark the connection as disposed so that
           // it gets destroyed by the pool and is never used again. If we don't do this and
           // return the connection to the pool, it will be useless until the current operation
           // that timed out, finally finishes.
-          this.connection.__knex__disposed = error;
+          connection.__knex__disposed = error;
```

Comment:
I'd like to do a similar change for `.__knex_disposed`, but that one actually affects behavior, so I left it for a different PR at another time